ext/js/language/ja/text-transformations.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {basicTextTransformationOptions} from '../text-transformations.js';
+import {convertAlphabeticToKana} from './japanese-wanakana.js';
+import {collapseEmphaticSequences, convertHalfWidthKanaToFullWidth, convertHiraganaToKatakana, convertKatakanaToHiragana, convertNumericToFullWidth} from './japanese.js';
+
+/** @type {import('language').TextTransformation<boolean[]>}*/
+const collapseEmphaticSequencesTransform = {
+    id: 'collapseEmphaticSequences',
+    name: 'Collapse emphatic character sequences',
+    description: 'すっっごーーい → すっごーい / すごい',
+    options: [
+        ['false', 'Disabled', [[false, false]]],
+        ['true', 'Collapse into single character', [[false, false], [true, false]]],
+        ['full', 'Remove all characters', [[false, false], [true, false], [true, true]]]
+    ],
+    transform: (str, setting, sourceMap) => {
+        const [collapseEmphatic, collapseEmphaticFull] = setting;
+        if (collapseEmphatic) {
+            str = collapseEmphaticSequences(str, collapseEmphaticFull, sourceMap);
+        }
+        return str;
+    }
+};
+
+/** @type {import('language').TextTransformation<any>[]} */
```

Comment:
I'll look a bit more deeply into this, but the `any` type should never be explicitly used. At best, there is an alias `SafeAny`, which basically just means that "okay we thought about this and there's not a better way", but I'll see if there's a better way of describing some of these.